contracts/cw-ics721-bridge/src/error.rs

```diff
@@ -0,0 +1,51 @@
+use cosmwasm_std::StdError;
+use cw_utils::ParseReplyError;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ContractError {
+    #[error(transparent)]
+    Std(#[from] StdError),
+
+    #[error("unauthorized")]
+    Unauthorized {},
+
+    #[error("only unordered channels are supported")]
+    OrderedChannel {},
+
+    #[error("invalid IBC channel version - got ({actual}), expected ({expected})")]
+    InvalidVersion { actual: String, expected: String },
+
+    #[error("ICS 721 channels may not be closed")]
+    CantCloseChannel {},
+
+    #[error("unrecognised class ID")]
+    UnrecognisedClassId {},
+
+    #[error("class ID already exists")]
+    ClassIdAlreadyExists {},
+
+    #[error("unrecognised reply ID")]
+    UnrecognisedReplyId {},
+
+    #[error(transparent)]
+    ParseReplyError(#[from] ParseReplyError),
+
+    #[error("must provide same number of token IDs and URIs")]
+    ImbalancedTokenInfo {},
+
+    #[error("unexpected uri for classID {class_id} - got ({actual:?}), expected ({expected:?})")]
+    ClassUriClash {
```

Comment:
Should this be called `ClassIdClash`? Don't see `class_uri` referenced anywhere in the comments or arguments for this error type.

Reply:
This is a great one to flag. I throw this error if we receive a NFT over IBC with a different `classUri` than previous NFTs with that `classId`.

https://github.com/public-awesome/ics721/blob/0bd6b6e995263675f03a4547833f4f71e2edc4f9/contracts/cw-ics721-bridge/src/contract.rs#L216-L220

The result of this is that we effectively say "the first NFT of this collection sets what metadata we expect, future ones are wrong if they disagree." Would love thoughts on how to do this better. Maybe we should just silence the error? Maybe we should override the previous one?

Reply:
Ok, this is a good @humanalgorithm question.

Not sure...

Reply:
Ok, had a good long chat about this. Seems the simplest thing to do is just silence the error.

Saves us having to worry about a lot of things. Would be good to leave a solid comment about it though. : )

Reply:
Cool! Will do. Thanks for all the thoughts.

Reply:
Done: https://github.com/public-awesome/ics721/pull/22/commits/dd3452f437b626b75cca783c5aad7277885c2d18